front/src/app/core/login/services/auth.service.ts

```diff
@@ -0,0 +1,407 @@
+                  email: result.user.email,
+                  displayName: result.user.displayName,
+                  photoURL: result.user.photoURL
+                });
+              } catch (error) {
+                console.error('Error during backend operations:', error);
+              }
+            }
+
+            window.history.replaceState({}, document.title, '/');
+            this.router.navigate(['/']);
+          })
+          .catch((error) => {
+            console.error('Connection error:', error);
+          });
+      }
+    }
+  }
+
+  private async sendTokenToBackend(token: string) {
+    try {
+      return await firstValueFrom(
+        this.http.post(`${environment.apiUrl}/auth/login`, { idToken: token }, { withCredentials: true })
+      );
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  private async saveUserToBackend(user: Partial<User>) {
+    if (!user?.uid) return;
+
+    try {
+      return await firstValueFrom(
+        this.http.post(`${environment.apiUrl}/auth`, user, { withCredentials: true })
+      );
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  async logout() {
+    await signOut(this.auth);
+    try {
+      await firstValueFrom(
+        this.http.post(`${environment.apiUrl}/auth/logout`, {}, { withCredentials: true })
+      );
+    } catch (error) {
+      console.error('Error during logout:', error);
+    }
+
+    this.userState.clearUser();
+
+    window.history.replaceState({}, document.title, '/');
+    this.user$.next(null);
+    this.router.navigate(['/']);
+  }
+
+  async getIdToken(forceRefresh = true): Promise<string | null> {
+    try {
+      if (!this.auth.currentUser) {
+        return null;
+      }
+
+      const token = await this.auth.currentUser.getIdToken(forceRefresh);
+      await this.sendTokenToBackend(token);
+
+      return token;
+    } catch (error) {
+      return null;
+    }
+  }
+
+  private async fetchUserData(uid: string): Promise<User | null> {
+    try {
+      return await firstValueFrom(
+        this.http.get<User>(`${environment.apiUrl}/auth/user/${uid}`, { withCredentials: true })
+      );
+    } catch (error) {
+      return null;
+    }
+  }
+
+  public openDialog(component: any, config: any) {
+    return this.dialog.open(component, config);
+  }
+
+  async processInvitations(user: FirebaseUser): Promise<void> {
+    if (!user || !user.email) return;
+
+    try {
+      await firstValueFrom(
+        this.http.post(
+          `${environment.apiUrl}/public/invitations/process`,
+          {
+            email: user.email.toLowerCase(),
+            uid: user.uid
+          }
+        )
+      );
+    } catch (error) {
+      console.error('Error processing invitations:', error);
+    }
+  }
+
+  getToken(): Observable<string | null> {
+    const user = this.currentUserSubject.value;
+    if (user && user.token) {
+      return of(user.token);
+    }
+
+    const token = localStorage.getItem('token');
+    return of(token);
```

Comment:
todo: ne pas utiliser `currentUserSubject.value` car c'est une mauvaise pratique (tu casses la chaîne de réactivité)

```suggestion
    return this.currentUserSubject.pipe(
        map(user => user?.token ?? localStorage.getItem('token')),
    );
```